internal/cmd/cli/apply/apply.go

```diff
@@ -227,24 +241,51 @@ func Dir(ctx context.Context, client Getter, name, baseDir string, opts *Options
 	}
 
 	if opts.Output == nil {
-		err = save(client, def, scans...)
+		err = save(ctx, client, def, opts, scans...)
 	} else {
 		_, err = opts.Output.Write(b)
 	}
 
 	return err
 }
 
-func save(client Getter, bundle *fleet.Bundle, imageScans ...*fleet.ImageScan) error {
+func pushOCIManifest(ctx context.Context, bundle *fleet.Bundle, opts *Options) (string, error) {
+	manifest := manifest.FromBundle(bundle)
+	manifestID, err := manifest.ID()
+	if err != nil {
+		return "", err
+	}
+	ociOpts := ociutils.OCIOpts{
+		URL:             opts.OCIRegistry.URL,
+		Username:        opts.OCIRegistry.Username,
+		Password:        opts.OCIRegistry.Password,
+		BasicHTTP:       opts.OCIRegistry.BasicHTTP,
+		InsecureSkipTLS: opts.OCIRegistry.InsecureSkipTLS,
+	}
+	oci := ociutils.NewOCIUtils()
+	err = oci.PushManifest(ctx, ociOpts, manifestID, manifest)
+	if err != nil {
+		return "", err
+	}
+	return manifestID, nil
+}
+
+func save(ctx context.Context, client Getter, bundle *fleet.Bundle, opts *Options, imageScans ...*fleet.ImageScan) error {
 	c, err := client.Get()
 	if err != nil {
 		return err
 	}
 
 	obj, err := c.Fleet.Bundle().Get(bundle.Namespace, bundle.Name, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		if _, err = c.Fleet.Bundle().Create(bundle); err != nil {
-			return err
+		if opts.OCIRegistry.URL != "" {
```

Comment:
nit: I feel like this function does to much. I would break it up into saveOCI, saveBundle, saveImagescan or so.
However, I tend to write funcs that  are too small for Go 🤷 

Well, I guess the logic is still readable and straight forward.